api/public/v2/flag/views.py

```diff
@@ -15,11 +21,24 @@ class FlagViewSet(viewsets.GenericViewSet, mixins.ListModelMixin, RepoPropertyMi
     lookup_field = "flag_name"
     queryset = RepositoryFlag.objects.none()
 
-    def get_queryset(self):
-        return self.repo.flags.all()
+    def get_queryset(self) -> QuerySet:
+        results = [
+            {"flag_name": f.flag_name, "coverage": 0} for f in self.repo.flags.all()
+        ]
+        try:
+            report = self.get_commit().full_report
+            if not report:
```

Comment:
just curious, when would we throw a NotFound vs. return None?

Reply:
this line would return the list of flags without the coverage information for each flag if the coverage report of the head commit is not yet processed.
in general imo we should throw 404 if the requested data is not found (eg in the param you have a repo/owner that doesn't exist) so there's nothing to return. And return null, I don't think we would return that as a response to the entire request, rather the response object has some partial data is not available.

Reply:
Gotcha! Makes sense to me, thanks for the context